src/hotspot/share/cds/aotArtifactFinder.cpp

```diff
@@ -73,6 +74,7 @@ void AOTArtifactFinder::find_artifacts() {
   // Note, if a class is not excluded, it does NOT mean it will be automatically included
   // into the AOT cache -- that will be decided by the code below.
   SystemDictionaryShared::finish_exclusion_checks();
+  AOTReferenceObjSupport::init_keep_alive_objs_table();
```

Comment:
This should be guarded with `INCLUDE_CDS_JAVA_HEAP` somehow. I guess the cleanest way would be to use `NOT_CDS_JAVA_HEAP_RETURN` in `aotReferenceObjSupport.hpp`

Reply:
I added `NOT_CDS_JAVA_HEAP_RETURN` to the declaration of `init_keep_alive_objs_table`. All the other functions are called only inside `INCLUDE_CDS_JAVA_HEAP` blocks, so I didn't change them to avoid cluttering the code.